src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,503 @@
+
+    modalRef.componentInstance.onConfirmExtensionCallBack.subscribe((isNotifyDeadlines: boolean) => {
+      this.handleDeleteDeadlines(selectedStudents, selectedInstructors, isNotifyDeadlines);
+      modalRef.close();
+    });
+  }
+
+  private handleCreateDeadlines(selectedStudents: StudentExtensionTableColumnModel[],
+    selectedInstructors: InstructorExtensionTableColumnModel[],
+    extensionTimestamp: number, isNotifyDeadlines: boolean): void {
+
+    const request: FeedbackSessionUpdateRequest = {
+      studentDeadlines: this.getUpdatedDeadlines(selectedStudents, extensionTimestamp, true),
+      instructorDeadlines: this.getUpdatedDeadlines(selectedInstructors, extensionTimestamp, false),
+      ...this.feedbackSessionDetails,
+    };
+
+    this.isSubmittingDeadlines = true;
+    this.feedbackSessionsService.updateFeedbackSession(this.courseId, this.feedbackSessionName, request,
+      isNotifyDeadlines)
+        .pipe(finalize(() => { this.isSubmittingDeadlines = false; }))
+        .subscribe(() => {
+          this.loadFeedbackSessionAndIndividuals();
+          this.statusMessageService.showSuccessToast(
+          `Extension(s) for ${selectedStudents.length} student(s) and ${selectedInstructors.length} instructor(s)
+           have been successful!`);
+        }, (resp: ErrorMessageOutput) => {
+         this.statusMessageService.showErrorToast(resp.error.message);
+        });
+  }
+
+  private getUpdatedDeadlines(selectedIndividuals: StudentExtensionTableColumnModel[] |
+    InstructorExtensionTableColumnModel[], extensionTimestamp: number, isStudent: boolean): Record<string, number> {
+      let record: Record<string, number> = {};
+      if (isStudent) {
+        record = { ...this.studentDeadlines };
+      } else {
+        record = { ...this.instructorDeadlines };
+      }
+
+      selectedIndividuals.forEach((x) => { record[x.email] = extensionTimestamp; });
+      return record;
+  }
+
+  private handleDeleteDeadlines(selectedStudents: StudentExtensionTableColumnModel[],
+    selectedInstructors: InstructorExtensionTableColumnModel[], isNotifyDeadlines: boolean): void {
+
+    const request: FeedbackSessionUpdateRequest = {
+      studentDeadlines: this.getDeletedDeadlines(selectedStudents, true),
+      instructorDeadlines: this.getDeletedDeadlines(selectedInstructors, false),
+      ...this.feedbackSessionDetails,
+    };
+
+    this.isSubmittingDeadlines = true;
+    this.feedbackSessionsService.updateFeedbackSession(this.courseId, this.feedbackSessionName, request,
+      isNotifyDeadlines)
+    .pipe(finalize(() => { this.isSubmittingDeadlines = false; }))
+    .subscribe(() => {
+      this.loadFeedbackSessionAndIndividuals();
+      this.statusMessageService.showSuccessToast(
+      `Deletion of extension(s) for ${selectedStudents.length} student(s) and ${selectedInstructors.length} `
+      + 'instructor(s) have been successful!');
```

Comment:
This probably should be indented

Reply:
Recommend to just configure the IDE/Editor you are using to auto fix it:)